common/crypto/platform/openssl/crypto.cpp

```diff
@@ -430,6 +439,97 @@ expected::ExpectedString SignRawData(
 	return Sign(private_key_path, shasum);
 }
 
+const size_t ecdsa256curveBits = 256;
+const size_t ecdsa256keySize = 32;
+
+// Try and decode the keys from pure binary, assuming that the points on the
+// curve (r,s), have been concatenated together (r || s), and simply dumped to
+// binary. Which is what we did in the `mender-artifact` tool.
+// (See MEN-1740) for some insight into previous issues, and the chosen fix.
+template <BIGNUM *DecoderFn(const unsigned char *, int, BIGNUM *)>
+static expected::ExpectedBytes TryASN1EncodeMenderCustomBinaryECFormat(
+	const vector<uint8_t> &signature, const mender::sha::SHA &shasum) {
+	// Verify that the marshalled keys match our expectation
+	const size_t assumed_signature_size {2 * ecdsa256keySize};
+	if (signature.size() > assumed_signature_size) {
+		return expected::unexpected(MakeError(
+			SetupError,
+			"Unexpected size of the signature for ECDSA. Expected 2*32 bytes. Got: "
+				+ to_string(signature.size())));
+	}
+	auto ecSig = unique_ptr<ECDSA_SIG, void (*)(ECDSA_SIG *)>(ECDSA_SIG_new(), ECDSA_SIG_free);
+	if (ecSig == nullptr) {
+		return expected::unexpected(MakeError(
+			SetupError,
+			"Failed to allocate the structure for the ECDSA signature" + GetOpenSSLErrorMessage()));
+	}
+
+	auto r = unique_ptr<BIGNUM, void (*)(BIGNUM *)>(
+		DecoderFn(signature.data(), ecdsa256keySize, NULL), BN_free);
+	if (r == NULL) {
+		return expected::unexpected(MakeError(
+			SetupError,
+			"Failed to extract the r(andom) part from the ECDSA signature in the binary representation"
+				+ GetOpenSSLErrorMessage()));
+	}
+	auto s = unique_ptr<BIGNUM, void (*)(BIGNUM *)>(
+		DecoderFn(signature.data() + ecdsa256keySize, ecdsa256keySize, NULL), BN_free);
+	if (s == NULL) {
+		return expected::unexpected(MakeError(
+			SetupError,
+			"Failed to extract the s(ignature) part from the ECDSA signature in the binary representation"
+				+ GetOpenSSLErrorMessage()));
+	}
+
+	// Set the r&s values in the SIG struct
+	// r & s now owned by ecSig
+	int ret {ECDSA_SIG_set0(ecSig.get(), r.release(), s.release())};
+	if (ret != OPENSSL_SUCCESS) {
+		return expected::unexpected(MakeError(
+			SetupError,
+			"Failed to set the signature parts in the ECDSA structure" + GetOpenSSLErrorMessage()));
+	}
+
+	/* Allocate some array guaranteed to hold the DER-encoded structure */
+	vector<uint8_t> arr(256);
```

Comment:
Could we give a name to the value?

Reply:
`a`? :laughing:

Reply:
After a long weekend, I've settled on `der_encoded_byte_array`

Reply:
ahh, hahaha.